crates/prune/src/pruner.rs

```diff
@@ -31,6 +31,13 @@ use tracing::{debug, error, instrument, trace};
 /// and `false` if there's more data to prune in further runs.
 pub type PrunerResult = Result<bool, PrunerError>;
 
+/// Result of part pruning.
+///
+/// Returns `true` if pruning has been completed up to the target block,
+/// and `false` if there's more data to prune in further runs. Also returns number of entries pruned
+/// (deleted from the database).
+type PrunePartResult = Result<(bool, usize), PrunerError>;
```

Comment:
I'm not a fan of these primitive tuple return values, especially for bools, 
can we replace the bool with an enum?

Reply:
agree, introduced a enum with two variants